src/fastoad/model_base/openmdao/tests/test_group.py

```diff
@@ -0,0 +1,179 @@
+    assert isinstance(problem.model.group.nonlinear_solver, om.NonlinearBlockGS)
+    assert problem.model.group.nonlinear_solver.options["maxiter"] == 100
+    assert problem.model.group.nonlinear_solver.options["iprint"] == 3
+
+    # -------------------------------------------------------------------------
+    problem = om.Problem()
+    with pytest.raises(ValueError) as err:
+        problem.model.add_subsystem("group", Group2(linear_solver=True))
+    assert 'please use "use_solvers=True" and "nonlinear_solver=False"' in err.value.args[0]
+
+    with pytest.raises(ValueError) as err:
+        problem.model.add_subsystem("group", Group2(nonlinear_solver=True))
+    assert 'please use "use_solvers=True" and "linear_solver=False"' in err.value.args[0]
+
+
+def test_cycle_group_with_default_solver_options():
+    class Group3(
+        CycleGroup,
+        default_linear_solver="om.LinearBlockGS",
+        default_nonlinear_solver="om.BroydenSolver",
+        default_linear_options={"iprint": 2},
+        default_nonlinear_options={"maxiter": 200, "iprint": -1},
+    ):
+        pass
+
+    # -------------------------------------------------------------------------
+    problem = om.Problem()
+    problem.model.add_subsystem("group", Group3())
+    problem.setup()
+
+    assert isinstance(problem.model.group.linear_solver, om.LinearBlockGS)
+    assert isinstance(problem.model.group.nonlinear_solver, om.BroydenSolver)
+    assert problem.model.group.linear_solver.options["iprint"] == 2
+    assert problem.model.group.nonlinear_solver.options["maxiter"] == 200
+    assert problem.model.group.nonlinear_solver.options["iprint"] == -1
+
+    # -------------------------------------------------------------------------
+    problem = om.Problem()
+    problem.model.add_subsystem(
+        "group",
+        Group3(
+            nonlinear_solver="om.NewtonSolver",
+            linear_options={"iprint": 0},
+            nonlinear_options={"iprint": 1},
+        ),
+    )
+    problem.setup()
+
+    assert isinstance(problem.model.group.linear_solver, om.LinearBlockGS)
+    assert isinstance(problem.model.group.nonlinear_solver, om.NewtonSolver)
+    assert problem.model.group.linear_solver.options["iprint"] == 0
+    assert problem.model.group.nonlinear_solver.options["maxiter"] == 200
+    assert problem.model.group.nonlinear_solver.options["iprint"] == 1
+
+    # -------------------------------------------------------------------------
+    problem = om.Problem()
+    problem.model.add_subsystem(
+        "group",
+        Group3(
+            linear_solver="om.DirectSolver",
+            nonlinear_solver=False,
+            linear_options={"iprint": 3},
+        ),
+    )
+    problem.setup()
+
+    assert isinstance(problem.model.group.linear_solver, om.DirectSolver)
+    assert isinstance(problem.model.group.nonlinear_solver, om.NonlinearRunOnce)
+    assert problem.model.group.linear_solver.options["iprint"] == 3
```

Comment:
Do you think it would be useful to add an example that uses the configuration file to modify those solver options and types ?

Reply:
Good point. It would have to be in integration tests, but it would be worth adding.

Reply:
Done in a7c0ff44a0fc0f1f0ff1ed3425c741f400253d32.

I took the occasion to remove some integration tests, to speed up the test workflows. CS25 has them all anyway.